### PR TITLE
Move KeyVault to own package.

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR moves the hosting extensions for Azure Key Vault into their own assembly. They don't need to be in the core. I've also added a README as a bit of a format check.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3120)

### CODE REVIEW

**Review by davidfowl — APPROVED** [review]

File: src/Aspire.Hosting.Azure.KeyVault/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+var builder = DistributedApplication.CreateBuilder(args);
+builder.AddAzureProvisioner();
+
+var webhookSigningSharedSecret = builder.AddParameter("webhooksecret", secret: true);
+
+var keyVault = builder.AddAzureKeyVault("mykeyvault", (_, construct, kv) => {
+
+  // Create a secret and assign an parameter resource to its value.
+  var secret = new KeyVaultSecret(construct, "secret");
+  secret.AssignProperty(x => x.Properties.Value, webhookSigningSharedSecret);
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.KeyVault/README.md:55]: Oh is that how you do it!

File: src/Aspire.Hosting.Azure.KeyVault/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Inside the the implementation of the application that depends on Key Vault (MyApp in this case) add the `Aspire.Azure.Security.KeyVault` package and follow the instructions in that package README to use the connection string that was injected by the code above.
+
+### Customizing the Azure Key Vault resource
+
+The `builder.AddAzureKeyVault(...)` extension method has an overload that allows for customization of the Key Vault resource that is created. In the below example a .NET Aspire parameter is defined which is then assigned to the value of a Key Vault secret which is created at provisioning time.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+builder.AddAzureProvisioner();
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.KeyVault/README.md:47]:
```suggestion
builder.AddAzureProvisioning();
```

File: src/Aspire.Hosting.Azure.KeyVault/README.md (hunk 1 above)
Comment (mitchdenny) [reply src/Aspire.Hosting.Azure.KeyVault/README.md:55]: Oh did you ask and I not answer you :) Yeah, its some overload magic I added :)

File: src/Aspire.Hosting.Azure.KeyVault/Aspire.Hosting.Azure.KeyVault.csproj [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <IsPackable>true</IsPackable>
+    <PackageTags>aspire hosting azure</PackageTags>
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.KeyVault/Aspire.Hosting.Azure.KeyVault.csproj:6]: Why doesn't this have `keyvault` in it?

File: src/Aspire.Hosting.Azure.KeyVault/Aspire.Hosting.Azure.KeyVault.csproj [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    <Description>Azure resource types for .NET Aspire.</Description>
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.KeyVault/Aspire.Hosting.Azure.KeyVault.csproj:7]: This should say "Azure KeyVault".

**Pull request merged**